flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/slotpool/SlotPoolImpl.java

```diff
@@ -353,8 +346,9 @@ private void requestSlotFromResourceManager(
 	private void slotRequestToResourceManagerFailed(SlotRequestId slotRequestID, Throwable failure) {
 		final PendingRequest request = pendingRequests.getKeyA(slotRequestID);
 		if (request != null) {
-			if (isBatchRequestAndFailureCanBeIgnored(request, failure)) {
-				log.debug("Ignoring failed request to the resource manager for a batch slot request.");
+			if (isSlotRequestFailureCanBeIgnored(failure)) {
```

Comment:
This change would lead to a user visible behavior change that streaming tasks will not fail immediately on RM slot allocation failures. They will instead wait until the timeout and fail then. This is essential when later we have a job running both streaming and batch tasks so that the streaming task slot requests do not need to fail but can wait for the batch task slots to be released. But at the moment it seems to be a regression which slows down the failure detection and recovery.

Reply:
I would keep both streaming task slot request and batch task slot request at the moment.
In this way the behavior on RM slot allocation failure will remain unchanged and we do not need to start a community discussion on it right now.